libs/grok/src/main/java/org/elasticsearch/grok/GrokCaptureConfig.java

```diff
@@ -54,10 +54,10 @@ public GrokCaptureType type() {
         return type;
     }
 
-    Object extract(byte[] textAsBytes, Region region) {
+    Object extract(byte[] utf8Bytes, int offset, Region region) {
         for (int number : backRefs) {
             if (region.beg[number] >= 0) {
-                String matchValue = new String(textAsBytes, region.beg[number], region.end[number] - region.beg[number],
+                String matchValue = new String(utf8Bytes, offset + region.beg[number], region.end[number] - region.beg[number],
```

Comment:
Ah, I was thrown off by this initially.

The region starts at index `0` and we have to offset it just like we did the initial match because we are extracting from the original utf8Bytes.

Reply:
`region.end[number] - region.beg[number]` is a length so adding the offset would sort of break it.

Reply:
@nik9000 D'OH